packages/medusa/src/services/shipping-option.ts

```diff
@@ -335,6 +366,73 @@ class ShippingOptionService extends TransactionBaseService {
     })
   }
 
+  async createShippingMethods(
+    data: {
+      option: ShippingOption
+      data: Record<string, unknown>
+      config: CreateShippingMethodDto
+      price: number
+    }[]
+  ) {
+    return this.atomicPhase_(async (manager) => {
+      const methodsToCreate = data.map((methodData) => {
+        const { option, data, price, config } = methodData
+
+        const toCreate: Partial<ShippingMethod> = {
+          shipping_option_id: option.id,
+          data: data,
+          price: price,
+        }
+
+        if (
+          this.featureFlagRouter_.isFeatureEnabled(
+            TaxInclusivePricingFeatureFlag.key
+          )
+        ) {
+          if (typeof option.includes_tax !== "undefined") {
+            toCreate.includes_tax = option.includes_tax
+          }
+        }
+
+        if (config.order) {
+          toCreate.order_id = config.order.id
+        }
+
+        if (config.cart) {
+          toCreate.cart_id = config.cart.id
+        }
+
+        if (config.cart_id) {
+          toCreate.cart_id = config.cart_id
+        }
+
+        if (config.return_id) {
+          toCreate.return_id = config.return_id
+        }
+
+        if (config.order_id) {
+          toCreate.order_id = config.order_id
+        }
+
+        if (config.claim_order_id) {
+          toCreate.claim_order_id = config.claim_order_id
+        }
```

Comment:
thought(non-blocking): this is going to feel great to clean up when we get link modules 🤤